ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -123,7 +123,7 @@ public void run() {
             long minTxnIdSeenOpen = txnHandler.findMinTxnIdSeenOpen();
             final long cleanerWaterMark = minTxnIdSeenOpen < 0 ? minOpenTxnId : Math.min(minOpenTxnId, minTxnIdSeenOpen);
 
-            LOG.info("Cleaning based on min open txn id: " + minOpenTxnId);
+            LOG.info("Cleaning based on min open txn id: " + cleanerWaterMark);
```

Comment:
Do I understand correctly that this is the fix and the other things are just general improvements of the code?

Reply:
No, that's just the logging fix. The main fix is in TxnHandler, see deleteInvalidOpenTransactions

Reply:
Thx